src/psyclone/domain/lfric/kernel_arg_order.py

```diff
@@ -0,0 +1,1063 @@
+        <field_name>"_stencil_size", where <field_name> is the default
+        name of the field with this stencil. This argument will
+        contain the number of cells in the stencil.
+
+        :param meta_arg: the metadata associated with a field argument \
+            with a stencil access.
+        :type meta_arg: \
+            :py:class:`psyclone.domain.lfric.kernel.FieldArgMetadata`
+
+        '''
+        field_name = self._field_name(meta_arg)
+        self.arg_info.append(f"{field_name}_stencil_size")
+        self._arg_index += 1
+
+    def stencil_unknown_direction(self, meta_arg):
+        '''The field entry stencil access is of type XORY1D so add an
+        additional integer direction argument of kind i_def and with
+        intent in, with default name <field_name>"_direction", where
+        <field_name> is the default name of the field with this
+        stencil.
+
+        :param meta_arg: the metadata associated with a field argument \
+            with a xory1d stencil access.
+        :type meta_arg: \
+            :py:class:`psyclone.domain.lfric.kernel.FieldArgMetadata`
+
+        '''
+        field_name = self._field_name(meta_arg)
+        self.arg_info.append(f"{field_name}_direction")
+        self._arg_index += 1
+
+    def stencil_2d(self, meta_arg):
+        '''Add a 3D stencil dofmap array of type integer, kind i_def and
+        intent in. The dimensions are (number-of-dofs-in-cell,
+        max-branch-length, 4). The default name is
+        <field_name>"_stencil_dofmap", where <field_name> is the
+        default name of the field with this stencil.
+
+        :param meta_arg: the metadata associated with a field argument \
+            with a stencil access.
+        :type meta_arg: \
+            :py:class:`psyclone.domain.lfric.kernel.FieldArgMetadata`
+
+        '''
+        field_name = self._field_name(meta_arg)
+        self.arg_info.append(f"{field_name}_stencil_dofmap")
+        self._arg_index += 1
+
+    def stencil(self, meta_arg):
+        '''Add a 2D stencil dofmap array of type integer, kind i_def and
+        intent in. The dimensions are (number-of-dofs-in-cell,
+        stencil-size). The default name is
+        <field_name>"_stencil_dofmap, where <field_name> is the
+        default name of the field with this stencil.
+
+        :param meta_arg: the metadata associated with a field argument \
+            with a stencil access.
+        :type meta_arg: \
+            :py:class:`psyclone.domain.lfric.kernel.FieldArgMetadata`
+
+        '''
+        field_name = self._field_name(meta_arg)
+        self.arg_info.append(f"{field_name}_stencil_dofmap")
+        self._arg_index += 1
+
+    def banded_dofmap(self, function_space, cma_operator):
+        '''Adds a banded dofmap for the provided function space and cma
+        operator when there is an assembly cma kernel.
+
+        Include the column-banded dofmap, the list of offsets for the
```

Comment:
"...dofmap which is a list of offsets for the..." if that's right?

Reply:
No longer in this PR.